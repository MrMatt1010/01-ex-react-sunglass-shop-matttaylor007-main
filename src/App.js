import "./App.css";
import products from "./api/products.json";

const FilterBy = () => {
  return (
    <div className="filterBy">
      <label htmlFor="category">Filter by</label>
      <select id="category" defaultValue="all">
        <option value="all">All</option>
        <option value="accessories">Accessories</option>
        <option value="sunglasses">Sunglasses</option>
      </select>
    </div>
  );
};

const SortBy = () => {
  return (
    <div className="sortBy">
      <label htmlFor="sort">Sort by</label>
      <select id="sort" defaultValue="latest">
        <option value="latest">Latest</option>
        <option value="low">Price: Low</option>
        <option value="high">Price: High</option>
      </select>
    </div>
  );
};

const Product = ({ image, name, description }) => {
  return (
    <li className="product-grid-item">
      <img src={image} alt={name} />
      <h3>{name}</h3>
      <p>{description}</p>
      <button>Buy now</button>
    </li>
  );
};

const Products = () => {
  return (
    <ul className="product-grid">
      <Product
        image={products[0].images[0]}
        name={products[0].name}
        description={products[0].description}
      />
      <Product
        image={products[1].images[0]}
        name={products[1].name}
        description={products[1].description}
      />
      <Product
        image={products[2].images[0]}
        name={products[2].name}
        description={products[2].description}
      />
      <Product
        image={products[3].images[0]}
        name={products[3].name}
        description={products[3].description}
      />
      <Product
        image={products[4].images[0]}
        name={products[4].name}
        description={products[4].description}
      />
      <Product
        image={products[5].images[0]}
        name={products[5].name}
        description={products[5].description}
      />
      <Product
        image={products[6].images[0]}
        name={products[6].name}
        description={products[6].description}
      />
      <Product
        image={products[7].images[0]}
        name={products[7].name}
        description={products[7].description}
      />
      <Product
        image={products[8].images[0]}
        name={products[8].name}
        description={products[8].description}
      />
      <Product
        image={products[9].images[0]}
        name={products[9].name}
        description={products[9].description}
      />
      <Product
        image={products[10].images[0]}
        name={products[10].name}
        description={products[10].description}
      />
      <Product
        image={products[11].images[0]}
        name={products[11].name}
        description={products[11].description}
      />
    </ul>
  );
};

const App = () => {
  return (
    <div className="container">
      <h1>Sunglass Shop</h1>
      <div className="toolbar">
        <FilterBy />
        <SortBy />
      </div>
      <Products />
    </div>
  );
};

export default App;
