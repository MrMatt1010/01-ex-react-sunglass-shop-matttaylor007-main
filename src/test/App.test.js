import { render, cleanup } from "@testing-library/react";
import App from "../App";

afterEach(cleanup);

test("renders the Sunglass Shop page as expected", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
