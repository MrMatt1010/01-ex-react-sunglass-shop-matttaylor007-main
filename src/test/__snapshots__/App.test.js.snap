// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Sunglass Shop page as expected 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      Sunglass Shop
    </h1>
    <div
      class="toolbar"
    >
      <div
        class="filterBy"
      >
        <label
          for="category"
        >
          Filter by
        </label>
        <select
          id="category"
        >
          <option
            selected=""
            value="all"
          >
            All
          </option>
          <option
            value="accessories"
          >
            Accessories
          </option>
          <option
            value="sunglasses"
          >
            Sunglasses
          </option>
        </select>
      </div>
      <div
        class="sortBy"
      >
        <label
          for="sort"
        >
          Sort by
        </label>
        <select
          id="sort"
        >
          <option
            selected=""
            value="latest"
          >
            Latest
          </option>
          <option
            value="low"
          >
            Price: Low
          </option>
          <option
            value="high"
          >
            Price: High
          </option>
        </select>
      </div>
    </div>
    <ul
      class="product-grid"
    >
      <li
        class="product-grid-item"
      >
        <img
          alt="THE RONNIE MAC"
          src="https://files.stripe.com/links/fl_test_HG5Y5oY5y2SlpQMclkQ0rjXm"
        />
        <h3>
          THE RONNIE MAC
        </h3>
        <p>
          The sunglasses Ronnie Mac would wear if Ronnie Mac wore sunglasses.
        </p>
        <button>
          Buy now
        </button>
      </li>
      <li
        class="product-grid-item"
      >
        <img
          alt="THE BLACKING OUT"
          src="https://files.stripe.com/links/fl_test_MUQR3s4Rlmp8eJ8hA7JBIAYV"
        />
        <h3>
          THE BLACKING OUT
        </h3>
        <p>
          Wait? Hahaha we made these?! I don’t remember that.  They have an ANSI Z87+ safety rating, roomier fit, rubberized black frame, and redesigned ear and nose pieces?! Damn, we did a good job. Someone give me a raise. 
        </p>
        <button>
          Buy now
        </button>
      </li>
      <li
        class="product-grid-item"
      >
        <img
          alt="EXPERTS ONLY LOUNGER BOTTOMS"
          src="https://files.stripe.com/links/fl_test_OySmdVDzKgBc9b3gbKXn22cg"
        />
        <h3>
          EXPERTS ONLY LOUNGER BOTTOMS
        </h3>
        <p>
          “The best thing we’ve ever made.” - Description writer.
        </p>
        <button>
          Buy now
        </button>
      </li>
      <li
        class="product-grid-item"
      >
        <img
          alt="THE DOLCE VITA"
          src="https://files.stripe.com/links/fl_test_9bdFA0Nt7GJ9Q7SgThJoMvXr"
        />
        <h3>
          THE DOLCE VITA
        </h3>
        <p>
          This time we got it right. We’re living like in a dolce vita. Mmmm, gonna dream tonight. We’re dancing like in a dolce vita. With lights and music on. Our love is made in the dolce vita.
        </p>
        <button>
          Buy now
        </button>
      </li>
      <li
        class="product-grid-item"
      >
        <img
          alt="1993 NIGHT SHADES"
          src="https://files.stripe.com/links/fl_test_UwxwT7mgCWx4EEOHrd1XPjiS"
        />
        <h3>
          1993 NIGHT SHADES
        </h3>
        <p>
          Just be careful because it is hard to keep ladies from stealing these at the bar.
        </p>
        <button>
          Buy now
        </button>
      </li>
      <li
        class="product-grid-item"
      >
        <img
          alt="NIGHT FADE SCRUNCHIE"
          src="https://files.stripe.com/links/fl_test_bXsKkOXQ00MgPnpI5foyviy9"
        />
        <h3>
          NIGHT FADE SCRUNCHIE
        </h3>
        <p>
          The Optimal Blend of Style and Performance. Fluffy, brightly colored scrunchies that won’t tear your hair out like elastic hairbands or the stranger at the bus stop.
        </p>
        <button>
          Buy now
        </button>
      </li>
      <li
        class="product-grid-item"
      >
        <img
          alt="BLOCK HARD SUPER MESH HAT"
          src="https://files.stripe.com/links/fl_test_qYDrplzhQOR72n7dKJ3C48PQ"
        />
        <h3>
          BLOCK HARD SUPER MESH HAT
        </h3>
        <p>
          The only logical evolution of the trucker hat, we bring you the Super Mesh hat. Even less hat for your buck. Featuring 5 panels, FOUR of which are made of mesh.
        </p>
        <button>
          Buy now
        </button>
      </li>
      <li
        class="product-grid-item"
      >
        <img
          alt="BLOCK HARD BELT"
          src="https://files.stripe.com/links/fl_test_wgtK9fPWBkN9kKViN8WYdyv9"
        />
        <h3>
          BLOCK HARD BELT
        </h3>
        <p>
          Tight or loose, you get to choose! A necklace for your pants. The ultimate in pantsing defense.
        </p>
        <button>
          Buy now
        </button>
      </li>
      <li
        class="product-grid-item"
      >
        <img
          alt="DAY FADE SCRUNCHIE"
          src="https://files.stripe.com/links/fl_test_fyBXinaucwAuw1IrFLK0mKAX"
        />
        <h3>
          DAY FADE SCRUNCHIE
        </h3>
        <p>
          The Optimal Blend of Style and Performance. Fluffy, brightly colored scrunchies that won’t tear your hair out like elastic hairbands or the stranger at the bus stop.
        </p>
        <button>
          Buy now
        </button>
      </li>
      <li
        class="product-grid-item"
      >
        <img
          alt="THE GOLD STANDARD POLARIZED"
          src="https://files.stripe.com/links/fl_test_iUCrsMz2JliR0gCd2Wv7JkIz"
        />
        <h3>
          THE GOLD STANDARD POLARIZED
        </h3>
        <p>
          Gold, our primal instincts draw us to this element.
        </p>
        <button>
          Buy now
        </button>
      </li>
      <li
        class="product-grid-item"
      >
        <img
          alt="THE LAND LOCKED POLARIZED"
          src="https://files.stripe.com/links/fl_test_InvctxcO4PLS7cWEcDJd1nnB"
        />
        <h3>
          THE LAND LOCKED POLARIZED
        </h3>
        <p>
          No tortoises were harmed in the making of these incredible tortoiseshell Exciters.
        </p>
        <button>
          Buy now
        </button>
      </li>
      <li
        class="product-grid-item"
      >
        <img
          alt="THE 1993 2000"
          src="https://files.stripe.com/links/fl_test_5OpXfbOUjAp3LFSfrJuKlqwc"
        />
        <h3>
          THE 1993 2000
        </h3>
        <p>
          The 1993s are back and better than ever. Demand Respect and Authority™. In the courtroom. In the bedroom. In the bathroom. In your mugshot. In, the FUTURE.
        </p>
        <button>
          Buy now
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
